File: TP/SAST/hardcoded-secret.py
from flask import Flask, request

app = Flask(__name__)

USERNAME = "admin"
PASSWORD = "123456"

@app.route('/')
def home():
    return "Welcome to the application"

@app.route('/login', methods=['POST'])
def login():
    username = request.form.get('username')
    password = request.form.get('password')

    if username == USERNAME and password == PASSWORD:
        return "Login successful!"
    else:
        return "Invalid credentials!"

File: TP/SAST/hardcoded-secret.swift
import UIKit

class ViewController: UIViewController {

    let username = "admin"
    let password = "password"

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func loginButtonTapped(_ sender: UIButton) {
        let inputUsername = "admin"
        let inputPassword = "password"

        if inputUsername == username && inputPassword == password {
            print("Login successful")
        } else {
            print("Incorrect credentials")
        }
    }
}
